Register Triton scaled dot production in aten (#82509)

zrphercule opened the pull request:
Summary:
This diff creates a aten Triton path along with a dummy aten scaled dot attention operator

Pull Request resolved: https://github.com/pytorch/pytorch/pull/82509

Test Plan: CI, since the operator is dummy (not to-be registered in python), there is no way to test its correctness in this diff

Reviewed By: erichan1, ngimel

Differential Revision: D37897092

Pulled By: zrphercule

erichan1 reviewed (approved):
Looks good! Let's get Triton in. We really should refactor most of the BetterTransformer MHA though so we don't have to copy all over the place. But that's something for another day.

zrphercule commented:
@pytorchbot merge -g

zrphercule commented:
@pytorchbot merge

zrphercule committed:
Register Triton scaled dot production in aten (#82509) (#82509)

Summary:
This diff creates a aten Triton path along with a dummy aten scaled dot attention operator

Pull Request resolved: https://github.com/pytorch/pytorch/pull/82509

Pull Request resolved: https://github.com/pytorch/pytorch/pull/82758

Test Plan: CI, since the operator is dummy (not to-be registered in python), there is no way to test its correctness in this diff

Reviewed By: erichan1, ngimel

Pulled By:
zrphercule
zrphercule

Differential Revision: D37897092

fbshipit-source-id: 037c98afdc8a67364cf077d983e23b159841c234

zrphercule commented:
@pytorchbot merge
